apps/exchange/test/matching_engine_test.exs

```diff
@@ -656,4 +656,152 @@ defmodule MatchingEngineTest do
       assert total_active == ids
     end
   end
+
+  describe "Message bus" do
+    setup _context do
+      {:ok, %{order_book: Utils.empty_order_book()}}
+    end
+
+    test "Check if trade_executed event is correctly broadcasted", %{order_book: order_book} do
+      TestEventBus.flush()
+      order_1 = Utils.sample_order(%{size: 1000, price: 1000, side: :buy})
+      order_2 = Utils.sample_order(%{size: 1000, price: 1000, side: :sell})
+      order_1 = %Order{order_1 | trader_id: "alchemist0", order_id: "100"}
+      order_2 = %Order{order_2 | trader_id: "alchemist9"}
+      ids = ~w(100 9)
+      {_reply, _response, order_book} =
+        MatchingEngine.handle_call({:place_limit_order, order_1}, nil, order_book)
+      {_reply, _response, _order_book} =
+        MatchingEngine.handle_call({:place_limit_order, order_2}, nil, order_book)
```

Comment:
The `handle_call` function should only be used internally by the GenServer, instead you can call `MatchingEngine.place_limit_order`